### PR TITLE
GDTCORStorage: promise API wrapper

### PR DESCRIPTION
**Pull request by maksymmalyhin**
The next PR from promise refactoring series (b/160018519). 

- promise based API for `GDTCORFlatFileStorage` that will be used in `GDTCCTUploadOperation`

Tests to be fixed and updated in the next PRs.

#no-changelog

### CODE REVIEW

**Comment by maksymmalyhin** [comment]
cc @mikehaney24

File: GoogleDataTransport/GDTCORLibrary/Private/GDTCORUploadBatch.h
Comment (paulb777) [inline GoogleDataTransport/GDTCORLibrary/Private/GDTCORUploadBatch.h:1]: In the rest of the repo, the Private folder is for internal headers used by other libraries in the repo.

Headers only needed by their own library are typically in the same folder as the sources.

We should consider making GDT consistent with that, either as new headers are added or as a follow on.

File: GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCORStorageProtocol.h
Comment (paulb777) [inline GoogleDataTransport/GDTCORLibrary/Public/GoogleDataTransport/GDTCORStorageProtocol.h:129]: Should be a minor version bump for this, but can wait until merge to master.

**Review by paulb777 — APPROVED** [review]

File: GoogleDataTransport/GDTCORLibrary/Private/GDTCORUploadBatch.h
Comment (maksymmalyhin) [reply GoogleDataTransport/GDTCORLibrary/Private/GDTCORUploadBatch.h:1]: I'll follow common private/internal headers rules for the future headers. I think the current ones can be moved as a part of #6383. I added a note to the ticket description.

**Pull request merged**
